### PR TITLE
fix(checkbox): missing focus indicator

### PR DESCRIPTION
**Pull request by crisbeto**
Adds a ripple when a checkbox is focused.

Referencing #421.

### CODE REVIEW

File: src/lib/checkbox/checkbox.ts [hunk 1]
```diff
@@ -69,7 +69,7 @@ export class MdCheckboxChange {
     '[class.md-checkbox-checked]': 'checked',
     '[class.md-checkbox-disabled]': 'disabled',
     '[class.md-checkbox-align-end]': 'align == "end"',
-    '[class.md-checkbox-focused]': 'hasFocus',
+    '[class.md-ripple-focused]': 'hasFocus',
```
Comment (jelbourn) [inline src/lib/checkbox/checkbox.ts]: Can you add a TODO to only show the focus style when it was caused by keyboard?

**Comment by jelbourn** [comment]
Just pulled this down to try it out. If I tab between checkboxes quickly, there's a trail of fading focus indicators left in my wake.

**Comment by crisbeto** [comment]
It's the same issue as [the one I mentioned here.](https://github.com/angular/material2/pull/1731#issuecomment-264630036)

**Comment by andrewseguin** [comment]
Please rebase

**Comment by crisbeto** [comment]
Closing and will re-do based on the current APIs.

**Pull request closed without merging**